server/api/db/sqldb/db.py

```diff
@@ -2687,17 +2678,18 @@ def refresh_project_summaries(
         session: Session,
         project_summaries: list[mlrun.common.schemas.ProjectSummary],
     ):
-        # Do the whole operation in a single transaction
-        with session.no_autoflush:
-            self._query(session, ProjectSummary).delete()
-            for project_summary_schema in project_summaries:
-                project_summary = ProjectSummary(
-                    project=project_summary_schema.name,
-                    summary=project_summary_schema.dict(),
-                    updated=datetime.now(timezone.utc),
-                )
-                session.add(project_summary)
-            session.commit()
+        project_summaries_to_update = []
+        for project_summary_schema in project_summaries:
+            query = self._query(
+                session, ProjectSummary, project=project_summary_schema.name
+            )
```

Comment:
because it can lead to a race condition, as @TomerShor mentioned earlier:
> We might still hit a race condition with the **delete** project flow, where the project is deleted after its summery was queried, and before the transaction ended. However, as we spoke offline, this might not be a real issue if the UI looks for the summery for each project that is returned in `list_projects`.

by checking that the project summary is not None after querying, this prevents the issue from occurring.
I will add a comment